pkg/components/kubernetesclient/kubernetesclientprovider/types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetesclientprovider
```

Comment:
Similar to previous changes, moving the functionality to access a Kubernetes client into a "provider". By deferring and standardizing the initialization that makes it easier to override in tests.